samples/cpp/text_generation/encrypted_model_causal_lm.cpp

```diff
@@ -39,18 +40,80 @@ ov::genai::Tokenizer decrypt_tokenizer(const std::string& models_path) {
     return ov::genai::Tokenizer(tok_model_str, tok_weights_tensor, detok_model_str, detok_weights_tensor);
 }
 
+static const char codec_key[] = {0x30, 0x60, 0x70, 0x02, 0x04, 0x08, 0x3F, 0x6F, 0x72, 0x74, 0x78, 0x7F};
+
+std::string codec_xor(const std::string& source_str) {
+    auto key_size = sizeof(codec_key);
+    int key_idx = 0;
+    std::string dst_str = source_str;
+    for (char& c : dst_str) {
+        c ^= codec_key[key_idx % key_size];
+        key_idx++;
+    }
+    return dst_str;
+}
+
+std::string encryption_callback(const std::string& source_str) {
+    return codec_xor(source_str);
+}
+
+std::string decryption_callback(const std::string& source_str) {
+    return codec_xor(source_str);
+}
+
+auto get_config_for_cache_encryption(const std::string& cache_dir, bool gpu) {
```

Comment:
```suggestion
auto get_config_for_cache_encryption(const std::string& cache_dir, bool is_gpu) {
```
similar to Python

Reply:
That raises a question about
> //set ov::CacheMode::OPTIMIZE_SIZE only for GPU to enable weightless cache

Why is it stated explicitly if `get_config_for_cache_encryption()` called only for GPU. If the reason is that other devices don't support it, you can sate that explicitly or remove the comment because it OV throws, users will notice anyway.

Reply:
agree
removed the comment